tests/ui/collapsible_str_replace.rs

```diff
@@ -0,0 +1,55 @@
+#![warn(clippy::collapsible_str_replace)]
+
+fn main() {
+    let misspelled = "hesuo worpd";
+
+    let p = 'p';
+    let s = 's';
+    let u = 'u';
+
+    // If there are consecutive calls to `str::replace` and none of the chars are variables,
+    // recommend `collapsible_str_replace`
+    let replacement = misspelled.replace('s', "l").replace('u', "l");
+    println!("{replacement}");
+
+    let replacement = misspelled.replace('s', "l").replace('u', "l").replace('p', "l");
+    println!("{replacement}");
+
+    // If the first argument to a single `str::replace` call is a slice and none of the chars
+    // are variables, recommend `collapsible_str_replace`
+    let replacement = misspelled.replace(&['s', 'u', 'p'], "l");
+    println!("{replacement}");
+
+    // If there are consecutive calls to `str::replace` and all or any chars are variables,
+    // recommend the fallback `misspelled.replace(&[s, u, p], "l")`
+    let replacement = misspelled.replace(s, "l").replace('u', "l");
+    println!("{replacement}");
+
+    let replacement = misspelled.replace(s, "l").replace('u', "l").replace('p', "l");
+    println!("{replacement}");
+
+    let replacement = misspelled.replace(s, "l").replace(u, "l").replace('p', "l");
+    println!("{replacement}");
+
+    let replacement = misspelled.replace(s, "l").replace(u, "l").replace(p, "l");
+    println!("{replacement}");
+
+    // If the first argument to a single `str::replace` call is a slice and one or more chars
+    // are variables, do not recommend `collapsible_str_replace`
+    let replacement = misspelled.replace(&['s', u, 'p'], "l");
```

Comment:
I think the comment, why this shouldn't be linted is wrong. If there's another `replace` behind this `replace` it could be added to the slice. But producing this suggestion might be hard. So this is not necessary for the first version of this lint. 

But yeah, this is not linted because there is only one `replace`, not because the pattern is a slice.